src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -2,36 +2,38 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
+
 using System.Collections.Immutable;
 using Microsoft.CodeAnalysis.Operations;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class BoundNode : IBoundNodeWithIOperationChildren
     {
-        ImmutableArray<BoundNode> IBoundNodeWithIOperationChildren.Children => this.Children;
+        ImmutableArray<BoundNode?> IBoundNodeWithIOperationChildren.Children => this.Children;
 
         /// <summary>
         /// Override this property to return the child bound nodes if the IOperation API corresponding to this bound node is not yet designed or implemented.
         /// </summary>
         /// <remarks>Note that any of the child bound nodes may be null.</remarks>
-        protected virtual ImmutableArray<BoundNode> Children => ImmutableArray<BoundNode>.Empty;
+        protected virtual ImmutableArray<BoundNode?> Children => ImmutableArray<BoundNode?>.Empty;
     }
 
     internal partial class BoundBadStatement : IBoundInvalidNode
     {
-        protected override ImmutableArray<BoundNode> Children => this.ChildBoundNodes;
+        protected override ImmutableArray<BoundNode?> Children => this.ChildBoundNodes!;
```

Comment:
>this.ChildBoundNodes!; [](start = 66, length = 22)

who do we know the suppression is okay?

Reply:
`ChildBoundNodes` is of type `ImmutableArray<BoundNode>`.  Immutable arrays act as covariant with respect to nullability, but the compiler doesn't know that.

---
In reply to: [386581038](https://github.com/dotnet/roslyn/pull/42068#discussion_r386581038) [](ancestors = 386581038)